js-core/homeworks/homework-4/src/main.js

```diff
@@ -0,0 +1,130 @@
+/*
+* Напишите функцию которая будет принимать 1 аргумент и в зависимости от типа аргумента
+* Если тип аргумента число или объект -> возвращать true
+* Если тип аргумента функция -> возвращать false
+* Если тип аргумента Строка и длина этой строки не равна 10 -> возвращать "длина вашей строки: <длина строки>
+* Если длина 10 -> 'you win'
+* */
+let objCheck = {
+    name: 'aloe'
+};
+let numbCkeck = 15;
+let stringCheck = 'aloe vera lorem ipsum';
+let stringTenCheck = 'aloe vera.';
+let funcCheck = function(){
+    return 'yes it is';
+};
+
+function check(param){
+    let numOrObject = typeof param === 'number' || typeof param === 'object';
+    let functionType = typeof param === 'function';
+    let stringType = typeof param === 'string' && param.length != 10;
+    let stingTenType = typeof param === 'string' && param.length === 10;
```

Comment:
stringTenType ? :)

such naming not really give us correct abstraction.

I guess better to name it something like

isStringWithLength10.

But I like your solution at all. What do you think maybe it worth to move to another function ?
```js 
typeof param === string
```
You can avoid DRY rule in other places